### DISCUSSION TITLE
feat: Add Typescript to codebase

### DISCUSSION THREAD

**Discussion by Carlos-err406** [open-webui/open-webui#2575] [General]
**Is your feature request related to a problem? Please describe.**
Despite the Svelte project being configured for TypeScript, with many components flagged using lang=ts, the utilization of types throughout the codebase is inconsistent. This leads to challenges in maintaining the code and results in numerous errors flagged by the language server, complicating development.

**Describe the solution you'd like**
To improve code quality and developer experience, I propose refactoring the existing code to ensure appropriate usage of TypeScript types wherever feasible. Additionally, incorporating type declarations for external libraries within package.json will enhance compatibility and reduce language server errors.

**Additional context**
I plan to use this project at my company and am ready to dedicate time to enhance it. If this aligns with the project's goals, I'm eager to help implement these improvements.

**Comment by tjbck** [comment]
Feel free to make a PR! Just a warning this project is still being actively developed and a lot of the code will be either replaced in favour of other implementations or removed entirely, so it's probably best if you focus on only the core part of the code. Good luck!

**Comment by Carlos-err406** [comment]
Thanks for the heads up! I'll be extra careful to make the changes without causing hiccups in the main development.

**Comment by MonsieurBibo** [comment]
Is this still up to date ? There are almost no more JS files, though there are still quite a lot of svelte components in JS...

**Comment by blujedis** [comment]
The code base is largely typed now it seems. However there are about 50 problems reported in my IDE (VS Code)

There are quite a few stores in ```/src/lib/stores/index.ts``` without a defined type. Just as an example you could update the following:

```ts
export const terminalServers = writable([]);
```
To: 

```ts
export const terminalServers = writable<{id: string; name: string; url: string; }>([]);
```

_NOTE: the above was a quick once over, could be other properties missing but you get the idea._ 

There are also quite a few types set to "any". Probably the result of converting the project and not having the time to comb through ever type. 

Anyway if the maintainers are interested in resolving some of these typing issues along with missing Aria attributes...I'd be happy to chip away at those. Let me know.
